### DISCUSSION TITLE
Optimizing the style refresh

### DISCUSSION THREAD

**Discussion by BurningTreeC** [TiddlyWiki/TiddlyWiki5#8105] [Ideas]
Hello,

I'm trying to optimize the styleRefresh by inserting each Stylesheet as a single `<style>` tag in the page `<head>` 
That works well and I can preserve the order of the stylesheets.
If we do so, we can for example split the base Stylesheet into chunks when we know that a part of the stylesheet is going to change and put that part of the stylesheet into a separate Stylesheet.

In my TiddlyFlex plugin I have stylesheets that change very very often and this would improve the styleRefresh cycle.
It wouldn't be needed anymore to add the whole stylesheet again if something in one of the stylesheets changes.
Also, I've read that browsers perform better if styles are put in different `<style>` tags rather than a single big one.

**Comment by pmario** [comment]
The last time I did have a closer look, all stylesheets where recreated with basically every keystroke. So having more of them imo will not really help.

**Reply by BurningTreeC** [reply]
Hi @pmario , I will add a PR soon where you can test

**Comment by Jermolene** [comment]
Hi @BurningTreeC I do think that would be a worthwhile improvement. At some point I did investigate it, and I can't quite recall what blocked me from completing it.

**Comment by BurningTreeC** [comment]
I've now added PR #8106 @pmario @Jermolene 
Beware, for new windows the handling is missing.
I'm about to add it.

**Reply by BurningTreeC** [reply]
The handling for new windows is added.

**Comment by BurningTreeC** [comment]
I've added PR #8118 which should be preferred over #8106
